onnx/reference/reference_evaluator.py

```diff
@@ -516,7 +516,15 @@ def run(self, output_names, feed_inputs: Dict[str, Any], attributes: Optional[Di
         # step 2: execute nodes
         for node in self.rt_nodes_:
             self._log(1, "%s(%s) -> %s", node.op_type, node.input, node.output)
-            inputs = [results[i] for i in node.input]
+            inputs = []
+            for i in node.input:
+                if i not in results:
+                    raise RuntimeError(
+                        f"Unable to find input {i!r} in known results {sorted(results)}, "
+                        f"self.rt_inits_ has {sorted(self.rt_inits_)}, "
+                        f"feed_inputs has {sorted(feed_inputs)}."
+                    )
+                inputs.append(results[i])
```

Comment:
Maybe do the check and follow with `inputs = [results[i] for i in node.input]` for readability? So one block check the results and the next block does the assignment

Reply:
I guess you could also do `try inputs = [results[i] ...] catch KeyError ...`

Reply:
try ... catch would lose the information about which i is not found. I used Justin's suggestion.